libs/contract/src/lib/bucket/+state/bucket.service.ts

```diff
@@ -119,28 +125,22 @@ export class BucketService extends CollectionService<BucketState> {
     return Promise.all(promises);
   }
 
-
-  async addTerm(titleId: string, parentTermId: string, avails: AvailsFilter) {
-
+  async addBatchTerms(availsResults: AddTermConfig[]) {
     const orgId = this.orgService.org.id;
-    const rawBucket = await this.getActive();
-    const bucket = createBucket({ id: orgId, ...rawBucket });
-
-    const term = createBucketTerm(avails);
-
-    const sale = bucket.contracts.find(contract => contract.titleId === titleId && contract.parentTermId === parentTermId);
-
-    if (sale) {
-      sale.terms.push(term);
-    } else {
-      const contract = createBucketContract({ titleId, parentTermId, terms: [term] });
-      bucket.contracts.push(contract);
-    }
-
-    if (rawBucket) {
-      this.update(orgId, bucket);
-    } else {
-      this.add(bucket);
+    const bucket = await this.getActive();
+    const contracts = bucket?.contracts ?? [];
+    for (const { titleId, parentTermId, avail } of availsResults) {
+      const term = createBucketTerm(avail);
+      const sale = contracts.find(c => c.titleId === titleId && c.parentTermId === parentTermId);
+      if (sale) {
+        sale.terms.push(term);
+      } else {
+        const contract = createBucketContract({ titleId, parentTermId, terms: [term] });
+        contracts.push(contract);
+      }
     }
+    bucket
```

Comment:
You should return here to be able to do `await` in front of call.
```suggestion
    return bucket
```

Reply:
it works well without an await, but I added it in for sake of doubt.